nixos/modules/services/x11/extra-layouts.nix

```diff
@@ -0,0 +1,232 @@
+        description = ''
+          The path to the xkb types file.
+          This file specifies the key types that can be associated with
+          the various keyboard keys.
+          It must contain a <literal>xkb_types { ... }</literal> block.
+        '';
+      };
+
+    };
+  };
+
+  patchIn = pkgs: name: layout: with layout;
+  ''
+      # install layout files
+      ${optionalString (compatFile   != null) "cp '${compatFile}'   'compat/${name}'"}
+      ${optionalString (geometryFile != null) "cp '${geometryFile}' 'geometry/${name}'"}
+      ${optionalString (keycodesFile != null) "cp '${keycodesFile}' 'keycodes/${name}'"}
+      ${optionalString (symbolsFile  != null) "cp '${symbolsFile}'  'symbols/${name}'"}
+      ${optionalString (typesFile    != null) "cp '${typesFile}'    'types/${name}'"}
+
+      # patch makefiles
+      for type in compat geometry keycodes symbols types; do
+        if ! test -f "$type/${name}"; then
+          continue
+        fi
+        test "$type" = geometry && type_name=geom || type_name=$type
+        ${pkgs.ed}/bin/ed -v $type/Makefile.am <<EOF
+      /''${type_name}_DATA =
+      a
+      ${name} \\
+      .
+      w
+      EOF
+        ${pkgs.ed}/bin/ed -v $type/Makefile.in <<EOF
+      /''${type_name}_DATA =
+      a
+      ${name} \\
+      .
+      w
+      EOF
+      done
+
+      # add model description
+      ${pkgs.ed}/bin/ed -v rules/base.xml <<EOF
+      /<\/modelList>
+      -
+      a
+      <model>
+        <configItem>
+          <name>${name}</name>
+          <_description>${layout.description}</_description>
+          <vendor>${layout.description}</vendor>
+        </configItem>
+      </model>
+      .
+      w
+      EOF
+
+      # add layout description
+      ${pkgs.ed}/bin/ed -v rules/base.xml <<EOF
+      /<\/layoutList>
+      -
+      a
+      <layout>
+        <configItem>
+          <name>${name}</name>
+          <_shortDescription>${name}</_shortDescription>
+          <_description>${layout.description}</_description>
+          <languageList>
+            ${lib.concatMapStrings (lang: "<iso639Id>${lang}</iso639Id>\n") layout.languages}
+          </languageList>
+        </configItem>
+        <variantList/>
+      </layout>
+      .
+      w
+      EOF
+  '';
+
+in
+
+{
+
+  ###### interface
+
+  options.services.xserver = {
+    extraLayouts = mkOption {
+      type = types.attrsOf (types.submodule layoutOpts);
+      default = {};
+      example = literalExample
+      ''
+        {
+          mine = {
+            description = "My custom xkb layout.";
+            languages = [ "eng" ];
+            symbolsFile = /path/to/my/layout;
+          };
+        }
+      '';
+      description = ''
+        Extra custom layouts that will be included in the xkb configuration.
+        Information on how to create a new layout can be found here:
+        https://www.x.org/releases/current/doc/xorg-docs/input/XKB-Enhancing.html#Defining_New_Layouts
```

Comment:
You need to use the xml link syntax here to have it be clickable in the manual

Reply:
Should be fixed now.